.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,328 @@
+            new_build_id=69
+          else
+            latest_version=$(echo "$response" | jq -r '.[0].name // empty')
+            
+            # Extract MAJOR, MINOR, and RELEASE from the latest version
+            IFS='.' read -r last_major last_minor last_release last_build <<< "$latest_version"
+            
+            # Compare with the current MAJOR, MINOR, and RELEASE
+            if [[ "$last_major" -eq "${{ env.MAJOR }}" && "$last_minor" -eq "${{ env.MINOR }}" && "$last_release" -eq "${{ env.RELEASE }}" ]]; then
+              # Increment the last number if they match
+              new_build_id=$((last_build + 1))
+            else
+              # Reset build number to 0 if the current version is different
+              new_build_id=0
+            fi
+          fi
+
+          echo "latest build number: ${latest_version:-'None found'}"
+          echo "new build number: ${new_build_id}"
+          build_id="${new_build_id}"
+        else
+          build_id="999${{ env.RUN }}"
+        fi
+        
+        echo "New Build ID: ${build_id}"
+        echo "BUILD_ID=${build_id}" >> $GITHUB_ENV
+        echo "::set-output name=BUILD_ID::${build_id}"
+      
+    - name: Get App Version
+      id: get_app_version
+      shell: bash
+      run: |
+        app_version="${{ env.MAJOR }}.${{ env.MINOR }}.${{ env.RELEASE }}.${{ env.BUILD_ID }}"
+        echo "App Version: ${app_version}"
+        echo "APP_VERSION=${app_version}" >> $GITHUB_ENV
+        echo "::set-output name=APP_VERSION::${app_version}"
+
+  build-macos-arm64:
+    runs-on: macos-latest
+    needs: get-version
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+
+      - name: define env variables
+        run: |
+          echo "APP_VERSION=${{needs.get-version.outputs.APP_VERSION}}" >> $GITHUB_ENV
+          echo "BUILD_ID=${{needs.get-version.outputs.BUILD_ID}}" >> $GITHUB_ENV
+
+      - name: Restore dependencies
+        run: dotnet restore --source https://ci.appveyor.com/nuget/ospsuite-utility --source https://ci.appveyor.com/nuget/ospsuite-bddhelper --source https://www.nuget.org/api/v2/
+
+      - name: Update version.h
+        run: |
+          LC_ALL=C sed -i '' -e "s@RELEASE 0@RELEASE ${{env.RELEASE}}@g" \
+                     -e "s@MAJOR 0@MAJOR ${{env.MAJOR}}@g" \
+                     -e "s@MINOR 0@MINOR ${{env.MINOR}}@g" \
+                     -e "s@BUILD 0@BUILD ${{ env.BUILD_ID }}@g" \
+                     src/OSPSuite.FuncParserNative/version.h
+
+      - name: Remove Native Project for macOS
+        run: dotnet sln OSPSuite.FuncParser.sln remove src/OSPSuite.FuncParserNative/OSPSuite.FuncParserNative.vcxproj
+
+      - name: Configure the native build (release)
+        run: cmake -BBuild/Release/x64/ -Hsrc/OSPSuite.FuncParserNative/ -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_ARCHITECTURES=arm64
```

Comment:
I think we don't want the ARCH hard coded right? We are deploying on ARM for ARM build and on intel for x64 build?